packages/next/client/script.tsx

```diff
@@ -78,14 +83,27 @@ const loadScript = (props: ScriptProps): void => {
   LoadCache.add(cacheKey)
 
   if (dangerouslySetInnerHTML) {
-    el.innerHTML = dangerouslySetInnerHTML.__html || ''
+    console.error(
+      "Using 'dangerouslySetInnerHTML' within <script> tags is not supported by Next.js."
+    )
   } else if (children) {
-    el.textContent =
-      typeof children === 'string'
-        ? children
-        : Array.isArray(children)
-        ? children.join('')
-        : ''
+    // Assign to el.textContent based on the type of children
```

Comment:
Make sure to add comments explaining the concatenation rule.

Reply:
I have added a comment now. Let me know if you think more details should be added to the explanation.